.github/workflows/on_build_do_test.yml

```diff
@@ -275,3 +281,67 @@ jobs:
             sha: ${{ needs.get-info.outputs.source-head-sha }}
             targetUrl: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
 
+  coverage:
+    needs: [ci-build-test-pr, db]
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    name: Publish Coverage Report
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Java
+        uses: ./.github/actions/setup-java
+
+      - uses: actions/download-artifact@v4
+        with:
+          pattern: surefire-test-report*
+
+      - name: Extract Maven Artifacts
+        uses: ./.github/actions/extract-maven-artifacts
+        with:
+          run-id: ${{ github.event.workflow_run.id }}
+
+      - name: Download Infinispan Server
+        id: dis
+        uses: ./.github/actions/download-infinispan
+        with:
+          run-id: ${{ github.event.workflow_run.id }}
+          github-token: ${{ github.token }}
+
+      - name: Extract Infinispan Source
+        run: |
+          mkdir test_dir && cd test_dir
+          unzip ${GITHUB_WORKSPACE}/infinispan-${{ steps.dis.outputs.server-version }}-src.zip
+          mv infinispan-${{ steps.dis.outputs.server-version }}-src infinispan
+
+      - name: Copy jacoco directories to Infinispan
+        run: |
+          cd test_dir/infinispan && mkdir jacoco
+          .github/scripts/merge-jacoco-files.sh "$GITHUB_WORKSPACE/test_dir/infinispan/jacoco/" "${GITHUB_WORKSPACE}/surefire-test-report*/"
+
+      - name: Coverage merge and report generation
+        id: coverage_merge
+        shell: bash
+        run: >
+          cd test_dir/infinispan &&
```

Comment:
We can avoid `cd` calls throughout the workflow by using `working-directory: ...`

https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#defaultsrunworking-directory

Reply:
@ryanemerson fixed